### PR TITLE
Pseudo-code for DataFrame

### PR DESCRIPTION
**Pull request by PProfizi**
Proposes pseudo-code for a DPF DataFrame.

### CODE REVIEW

File: src/ansys/dpf/post/dataframe.py [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+    except ImportError:
+        pass
+
+if PANDAS_PRESENT:
+    PandasDataFrameType = pd.DataFrame
+else:
+    PandasDataFrameType = TypeVar("PandasDataFrameType")
+
+display_width = 80
+display_max_colwidth = 16
```
Comment (MaxJPRey) [inline src/ansys/dpf/post/dataframe.py:28]:
```suggestion
DISPLAY_WIDTH = 80
DISPLAY_MAX_COLWIDTH = 16
```

Comment (MaxJPRey) [inline src/ansys/dpf/post/dataframe.py:28]: If you rename those global variables, you might have to rename them below -just to not forget about it 😃- .
```suggestion
DISPLAY_WIDTH = 80
DISPLAY_MAX_COLWIDTH = 16
```

File: src/ansys/dpf/post/dataframe.py [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+        if len(self) > 1:
+            raise NotImplementedError(
+                "This function requires for the DataFrame to contain only"
+                "one data entity."
+            )
+        df = pd.DataFrame(self.to_numpy(), columns=columns, index=self.index.mesh.ids)
+        df.name = self._fc[-1].name
+        return df
+
+    def to_numpy(self):
```
Comment (PProfizi) [inline src/ansys/dpf/post/dataframe.py:323]:
```suggestion
    def to_array(self):
```

File: temp/DPFDataframe.md [hunk 3]
```diff
@@ -0,0 +1,428 @@
...
+
+[ 100 rows x 18 (2 steps x 9 results) columns ]
+```
+With ``display_width=80`` (effective is 75) and ``display.min_colwidth = display.max_colwidth = 6``:
+```pycon
+>>> df
+    step     1                           ...     2                         
+  result     U                 S         ...     S               
+    comp    UX    UY    UZ   SXX   SYY   ...   SYY   SZZ   SXY   SYZ   SXZ
+ node ID
```
Comment (PProfizi) [inline temp/DPFDataframe.md:130]:
```suggestion
 node
```

**Pull request closed without merging**